src/sqlfluff/rules/L069.py

```diff
@@ -0,0 +1,175 @@
+"""Implementation of Rule L069."""
+
+from typing import Optional
+
+from sqlfluff.core.parser.segments.raw import CodeSegment
+from sqlfluff.core.rules.crawlers import SegmentSeekerCrawler
+from sqlfluff.core.rules import BaseRule, LintFix, LintResult, RuleContext
+from sqlfluff.utils.functional import sp, FunctionalContext
+from sqlfluff.core.rules.doc_decorators import (
+    document_configuration,
+    document_fix_compatible,
+    document_groups,
+)
+
+
+@document_groups
+@document_fix_compatible
+@document_configuration
+class Rule_L069(BaseRule):
+    """Use ``CAST`` instead of ``CONVERT`` or ``::``.
+
```

Comment:
Would be better to support
`consistent, shorthand, or cast` as options.
consistent is a very common approach from SQLFluff - meaning once we see a first type, `::` or `CAST` etc, then all the rest of the script gets changed to that style

Reply:
Alright. Thanks for suggesting that. Will look into it.

Reply:
Fixed by [ebeb4ad](https://github.com/sqlfluff/sqlfluff/pull/3747/commits/ebeb4ada735810c3d050d690d93e0d8fcc64d4fa)